File: leetcode/easy/longest_common_prefix.py
from typing import List


class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        result = strs[0]
        for word in strs[1:]:
            i = 0
            for i in range(len(result)):
                if i == len(word) or result[i] != word[i]:
                    break
                i += 1
            result = result[: i]
        return result


print(Solution().longestCommonPrefix(["flower","flow","flight"]))

File: leetcode/scala/easy/longest_common_prefix/longest_common_prefix.scala
import Math.min


object Solution {
    def longestCommonPrefix(strs: Array[String]): String = {
        strs.reduceLeft(_longestCommon)
    }

    private def _longestCommon(first: String, second: String): String = {
        var result = ""
        for(i <- 0 to min(first.length(), second.length())-1) {
            if(first.charAt(i) == second.charAt(i)) {
                result = result + first.charAt(i)
            } else return result
        }
        result
    }
}